### PR TITLE
Overhaul of Maven profile IDs

### PR DESCRIPTION
**Pull request by JonathanGiles**
This PR attempts to achieve two things:

1. It introduces an overhaul of Maven profile IDs, to try and enforce some degree of consistency. This makes working with the entire project simpler, in that users can more easily understand what the profiles do, and if the profile is relevant to their interests.
2. It splits libraries into subsets `libraries-data-plane-current`, `libraries-data-plane-track1`, `libraries-data-plane-spring`, `libraries-mgmt-plane-current`, and `libraries-mgmt-plane-track1`. Users can more easily enable / disable these profiles to focus on the subset of libraries they care about (although this is still quite coarse-grained). 

The result of these two changes is two-fold:

1. Building from the root of the repository does not by default result in all libraries being built. The current configuration is just that the `libraries-data-plane-current` and `libraries-data-plane-spring` libraries will be built, with others being enabled / disabled using build-time flags.
2. It is now more feasible to open the entire azure-sdk-for-java root pom.xml file within your preferred IDE, as the libraries presented will be substantially less than the whole set of libraries present in the mono-repo.

Here's a screenshot from IntelliJ showing the updated profile IDs:
image

### CODE REVIEW

File: sdk/storage/azure-storage-queue/pom.xml [hunk 1]
```diff
@@ -190,7 +190,7 @@
       </build>
     </profile>
     <profile>
-      <id>java9plus</id>
+      <id>requires-java-9-or-later</id>
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-queue/pom.xml:193]: This profile should be deleted

File: sdk/storage/azure-storage-file-share/pom.xml [hunk 2]
```diff
@@ -236,7 +236,7 @@
       </build>
     </profile>
     <profile>
-      <id>java9plus</id>
+      <id>requires-java-9-or-later</id>
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-share/pom.xml:239]: This profile should be deleted

File: sdk/storage/azure-storage-file-datalake/pom.xml [hunk 3]
```diff
@@ -263,7 +263,7 @@
       </build>
     </profile>
     <profile>
-      <id>java9plus</id>
+      <id>requires-java-9-or-later</id>
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-datalake/pom.xml:266]: This profile should be deleted

File: sdk/spring/spring-cloud-azure-starter-stream-eventhubs/pom.xml [hunk 4]
```diff
@@ -45,7 +45,7 @@
   <!-- Repositories definitions -->
   <repositories>
     <repository>
-      <id>ossrh</id>
+      <id>repo-ossrh</id>
```
Comment (alzimmermsft) [inline sdk/spring/spring-cloud-azure-starter-stream-eventhubs/pom.xml:48]: We should think of pulling these repositories into a Spring parent POM, and some other common configurations between Azure Spring libraries

File: sdk/parents/azure-data-sdk-parent/pom.xml [hunk 5]
```diff
@@ -500,6 +500,7 @@
       </build>
     </profile>
 
+    <!-- TODO (JG): Is this profile still necessary? -->
```
Comment (alzimmermsft) [inline sdk/parents/azure-data-sdk-parent/pom.xml:503]: Let's remove it and find out

File: sdk/applicationinsights/pom.xml [hunk 6]
```diff
@@ -9,8 +9,27 @@
   <packaging>pom</packaging>
   <version>1.0.0</version><!-- Need not change for every release-->
 
-  <modules>
-      <module>azure-resourcemanager-applicationinsights</module>
-      <module>microsoft-azure-applicationinsights-query</module>
-  </modules>
+  <profiles>
+    <profile>
+      <id>libraries-data-plane-current</id>
```
[the remark this answers is not in the document]
Comment (JonathanGiles) [reply sdk/applicationinsights/pom.xml:14]: We have a `libraries-data-plane-track1` for those libraries. It just appears I didn't use it correctly here - will update.

File: sdk/containerregistry/azure-containers-containerregistry/pom.xml [hunk 7]
```diff
@@ -115,7 +115,7 @@
   </dependencies>
   <profiles>
     <profile>
-      <id>java9-plus</id>
+      <id>requires-java-9-or-later</id>
```
[the remark this answers is not in the document]
Comment (JonathanGiles) [reply sdk/containerregistry/azure-containers-containerregistry/pom.xml:118]: Done

File: sdk/parents/azure-client-sdk-parent/pom.xml [hunk 8]
```diff
@@ -1627,8 +1568,9 @@
       </build>
     </profile>
 
+    <!-- TODO (JG): Is this profile still being used? -->
```
Comment (alzimmermsft) [inline sdk/parents/azure-client-sdk-parent/pom.xml:1571]: Yes, this is needed to add that additional system property to the build to allow libraries using SecurityManager to build in Java 18+

File: sdk/storage/azure-storage-queue/pom.xml (hunk 1 above)
Comment (JonathanGiles) [reply sdk/storage/azure-storage-queue/pom.xml:193]: Done

File: sdk/storage/azure-storage-file-share/pom.xml (hunk 2 above)
Comment (JonathanGiles) [reply sdk/storage/azure-storage-file-share/pom.xml:239]: done

File: sdk/storage/azure-storage-file-datalake/pom.xml (hunk 3 above)
Comment (JonathanGiles) [reply sdk/storage/azure-storage-file-datalake/pom.xml:266]: Done

File: sdk/parents/azure-data-sdk-parent/pom.xml (hunk 5 above)
Comment (JonathanGiles) [reply sdk/parents/azure-data-sdk-parent/pom.xml:503]: Done...let's see!

File: sdk/core/azure-core-http-jdk-httpclient/pom.xml [hunk 9]
```diff
@@ -222,8 +222,9 @@
     </profile>
 
     <!-- Override the parent client.pom.xml's java11+ profile to override target of base-compile execution from 1.8 to 11-->
+    <!-- TODO (JG): Should this LTS be moved up to 17? -->
```
[the remark this answers is not in the document]
Comment (JonathanGiles) [reply sdk/core/azure-core-http-jdk-httpclient/pom.xml:225]: The present activation is `[11,)`, which according to https://maven.apache.org/enforcer/enforcer-rules/versionRanges.html means we activate on 11+. Should we increase the activation range to 12+?

[the remark this answers is not in the document]
Comment (alzimmermsft) [reply sdk/core/azure-core-http-jdk-httpclient/pom.xml:225]: We should, and then we can remove all the JUnit test skips requiring Java 12+

https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-http-jdk-httpclient/src/test/java/com/azure/core/http/jdk/httpclient/RestProxyWithJdkHttpClientTests.java#L15

File: sdk/parents/azure-client-sdk-parent/pom.xml (hunk 8 above)
Comment (JonathanGiles) [reply sdk/parents/azure-client-sdk-parent/pom.xml:1571]: Will leave it in - thanks.

**Review by kushagraThapar — COMMENTED** [review]
Thanks @JonathanGiles , this LGTM from Cosmos point of view, pending live tests run. 
@FabianMeiswinkel  - can you pls review spark as well?
@trande4884 - pls review spring data cosmos, thanks!

File: sdk/cosmos/azure-cosmos-spark_3_2-12/pom.xml [hunk 10]
```diff
@@ -713,7 +713,7 @@
 
     <!-- Override the parent client.pom.xml's java11+ profile to override target of base-compile execution from 1.8 to 11-->
     <profile>
-      <id>java-lts</id>
+      <id>requires-java-lts</id>
```
[the remark this answers is not in the document]
Comment (JonathanGiles) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/pom.xml:716]: Replaced with `requires-java-17-or-later` everywhere.

File: sdk/core/azure-core-http-jdk-httpclient/pom.xml (hunk 9 above)
[the remark this answers is not in the document]
Comment (JonathanGiles) [reply sdk/core/azure-core-http-jdk-httpclient/pom.xml]: Done

File: sdk/spring/pom.xml [hunk 11]
```diff
@@ -67,12 +79,16 @@
         <module>spring-cloud-azure-feature-management</module>
         <module>spring-cloud-azure-feature-management-web</module>
         <module>spring-cloud-azure-starter-appconfiguration-config</module>
-        <module>azure-spring-data-cosmos</module>
```
Comment (trande4884) [inline sdk/spring/pom.xml:70]: Why was this removed?

File: sdk/spring/pom.xml [hunk 12]
```diff
@@ -127,7 +143,6 @@
...
```
Comment (trande4884) [inline sdk/spring/pom.xml:130]: Why was this removed?
